site/src/modules/resources/AgentLogs/useAgentLogs.test.tsx

```diff
@@ -0,0 +1,137 @@
+    const fetchSpy = jest.spyOn(API, "getWorkspaceAgentLogs");
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    renderUseAgentLogs(queryClient, "ready", { enabled: false });
+    expect(fetchSpy).not.toHaveBeenCalled();
+    expect(wsSpy).not.toHaveBeenCalled();
+  });
+
+  it("should return existing logs without network calls", async () => {
+    const queryClient = createTestQueryClient();
+    queryClient.setQueryData(
+      agentLogsKey(MockWorkspace.id, MockWorkspaceAgent.id),
+      generateLogs(5),
+    );
+    const fetchSpy = jest.spyOn(API, "getWorkspaceAgentLogs");
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    const { result } = renderUseAgentLogs(queryClient, "ready");
+    await waitFor(() => {
+      expect(result.current).toHaveLength(5);
+    });
+    expect(fetchSpy).not.toHaveBeenCalled();
+    expect(wsSpy).not.toHaveBeenCalled();
+  });
+
+  it("should fetch logs when empty", async () => {
+    const queryClient = createTestQueryClient();
+    const fetchSpy = jest
+      .spyOn(API, "getWorkspaceAgentLogs")
+      .mockResolvedValueOnce(generateLogs(5));
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    const { result } = renderUseAgentLogs(queryClient, "ready");
+    await waitFor(() => {
+      expect(result.current).toHaveLength(5);
+    });
+    expect(fetchSpy).toHaveBeenCalledWith(MockWorkspaceAgent.id);
+    expect(wsSpy).not.toHaveBeenCalled();
+  });
+
+  it("should fetch logs and connect to websocket when agent is starting", async () => {
+    const queryClient = createTestQueryClient();
+    const logs = generateLogs(5);
+    const fetchSpy = jest
+      .spyOn(API, "getWorkspaceAgentLogs")
+      .mockResolvedValueOnce(logs);
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    new WS(
+      `ws://localhost/api/v2/workspaceagents/${
+        MockWorkspaceAgent.id
+      }/logs?follow&after=${logs[logs.length - 1].id}`,
+    );
+    const { result } = renderUseAgentLogs(queryClient, "starting");
+    await waitFor(() => {
+      expect(result.current).toHaveLength(5);
+    });
+    expect(fetchSpy).toHaveBeenCalledWith(MockWorkspaceAgent.id);
+    expect(wsSpy).toHaveBeenCalledWith(MockWorkspaceAgent.id, {
+      after: logs[logs.length - 1].id,
+      onMessage: expect.any(Function),
+      onError: expect.any(Function),
+    });
+  });
+
+  it("update logs from websocket messages", async () => {
+    const queryClient = createTestQueryClient();
+    const logs = generateLogs(5);
+    jest.spyOn(API, "getWorkspaceAgentLogs").mockResolvedValueOnce(logs);
+    const server = new WS(
+      `ws://localhost/api/v2/workspaceagents/${
+        MockWorkspaceAgent.id
+      }/logs?follow&after=${logs[logs.length - 1].id}`,
+    );
+    const { result } = renderUseAgentLogs(queryClient, "starting");
+    await waitFor(() => {
+      expect(result.current).toHaveLength(5);
+    });
+    await server.connected;
```

Comment:
Is this a typo? This doesn't look like a function call, so I wouldn't expect `await` to do anything here

Reply:
Not a typo. It is a JS thing where you can have async getters. https://github.com/romgain/jest-websocket-mock?tab=readme-ov-file#the-ws-constructor

Reply:
Why? For me it looks ok 😆